### PR TITLE
php 8.2.2

### PR DESCRIPTION
**Pull request by derrabus**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by SMillerDev** [comment]
I guess there's no `_www` user on Ubuntu anymore.

**Comment by chenrui333** [comment]
on linux
```
  ==> /home/linuxbrew/.linuxbrew/Cellar/php/8.2.2/sbin/php-fpm -t
  [02-Feb-2023 18:39:38] ERROR: [pool www] cannot get uid for user '_www'
  [02-Feb-2023 18:39:38] ERROR: FPM initialization failed
```

**Comment by derrabus** [comment]
Isn't the user for web server processes named `www-data` on Ubuntu?

**Comment by SMillerDev** [comment]
Well, the tests are complaining about `_www`

**Comment by derrabus** [comment]
Yeah, but that's how that user is called on macOS, afaik. So the formula was using the wrong user all along. Question is, why does this fail now and didn't previously? 🤔 

I don't have access to a Ubuntu machine with Homebrew at the moment, so I cannot investigate. 😕

**Comment by chenrui333** [comment]
> I don't have access to a Ubuntu machine with Homebrew at the moment, so I cannot investigate. 😕

yeah, you can leverage github action runner, there are [some code](https://github.com/chenrui333/github-action-test/blob/main/.github/workflows/brew.yml#L64-L80) refs for this. I can also take a look.

**Comment by derrabus** [comment]
> yeah, you can leverage github action runner, there are [some code](https://github.com/chenrui333/github-action-test/blob/main/.github/workflows/brew.yml#L64-L80) refs for this.

Thank you for the pointer!

> I can also take a look.

Feel free to do so. I won't have time before Monday.

**Comment by chenrui333** [comment]
just need a permission when creating/deleting `_www` test user

**Comment by chenrui333** [comment]
8.2.2 introduced [a bug fix](https://github.com/php/php-src/commit/21d8980d274e38ea8f9987498dde5ddb1b4b2f90) for [Configuration test does not perform uid lookups](https://bugs.php.net/bug.php?id=68591&edit=1).

File: Formula/php.rb [hunk 1]
```diff
@@ -117,6 +116,10 @@ def install
     # sdk path or it won't find the headers
     headers_path = "=#{MacOS.sdk_path_if_needed}/usr" if OS.mac?
 
+    # `_www` no longer exists in ubuntu-22.04, use `www-data` instead
+    fpm_user = OS.mac? ? "_www", "www-data"
+    fpm_group = OS.mac? ? "_www", "www-data"
```
Comment (SMillerDev) [inline Formula/php.rb:121]:
```suggestion
    # `_www` only exists on macOS. 
    fpm_user = OS.mac? ? "_www" : "www-data"
    fpm_group = OS.mac? ? "_www" : "www-data"
```

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**